umpost/conversion_driver_esm1p5.py

```diff
@@ -188,25 +252,47 @@ def convert_esm1p5_output_dir(esm1p5_output_dir):
         atm_dir_contents, fields_file_name_pattern
     )
 
-    if len(atm_dir_fields_files) > 0:
-         # Run the conversion
-        convert_fields_file_list(atm_dir_fields_files, nc_write_dir)
-    else:
+    if len(atm_dir_fields_files) == 0:
         warnings.warn(
             f"No files matching pattern '{fields_file_name_pattern}' "
             f"found in {current_atm_output_dir.resolve()}. No files will be "
             "converted to NetCDF."
         )
 
-   
+        # TODO: Check a better way of doing this
+        return  [], [] # Don't try to run the conversion
+
+    succeeded, failed = convert_fields_file_list(
+        atm_dir_fields_files,
+        nc_write_dir
+    )
+
+    return succeeded, failed
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
     parser.add_argument(
         "current_output_dir", help="ESM1.5 output directory to be converted", type=str
     )
+    parser.add_argument("--quiet", "-q", action="store-true",
+                        help=(
+                            "Report only final exception type and message for allowed"
+                            "exceptions raised during conversion when flag is included."
+                            "Otherwise report full stack trace."
+                        )
+                        )
     args = parser.parse_args()
 
     current_output_dir = args.current_output_dir
-    convert_esm1p5_output_dir(current_output_dir)
+
+    succeeded, failed = convert_esm1p5_output_dir(current_output_dir)
+
+    # Report results to user
+    success_reports = format_successes(succeeded)
+    failure_reports = format_failures(failed, args.quiet)
+    for success_message in success_reports:
```

Comment:
It's a minor stylistic comment, but this could be `for success_message in format_successes(succeeded):` if using the generator approach.